### ISSUE TITLE
Can't import tools in provider from json file

### ISSUE THREAD

**Issue by orange-anjou** [promptfoo/promptfoo#1311]
**Describe the bug**
When trying to import a tool from a json file in the provider i get an error.

**To Reproduce**

Conf 1 : 
```yml
...
tools:
   - file://./get_current_weather.json
```

Conf 2 : 
```yaml
...
tools: [
   file://./get_current_weather.json
]

```

It does work when putting the json directly.

Tool content : 
```yaml
{
  "type": "function",
  "function": {
    "name": "get_current_weather",
    "description": "Get the current weather in a given location",
    "parameters": {
      "type": "object",
      "properties": {
        "location": {
          "type": "string",
          "description": "The city and state, e.g. San Francisco, CA"
        },
        "unit": {
          "type": "string",
          "enum": [
            "celsius",
            "fahrenheit"
          ]
        }
      },
      "required": [
        "location"
      ]
    }
  }
}
```

Error : 
```
API error: Invalid type for 'tools[0]': expected an object, but got a string instead., Type: invalid_request_error, Code: invalid_type

{
  "error": {
    "message": "Invalid type for 'tools[0]': expected an object, but got a string instead.",
    "type": "invalid_request_error",
    "param": "tools[0]",
    "code": "invalid_type"
  }
}
```

**Expected behavior**
A provider correctly configured with the tool.

**System information (please complete the following information):**

- Promptfoo version: 0.73.7

**Comment by typpo** [comment]
Implementing in #1342 and should be in the next release!

**Comment by typpo** [comment]
Released in 0.76.0
